hl/src/H5LTanalyze.l

```diff
@@ -138,31 +103,12 @@ OPQ_SIZE         {return token(OPQ_SIZE_TOKEN);}
 OPQ_TAG          {return token(OPQ_TAG_TOKEN);}
 
 [0-9]+           {
-                        if( is_str_size || (is_enum && is_enum_memb) ||
-                            is_opq_size || (asindex>-1 && arr_stack[asindex].is_dim) ||
-                            (csindex>-1 && cmpd_stack[csindex].is_field) ) {
-                            H5LTyylval.ival = atoi(yytext);
-                            return NUMBER;
-                        } else
-                            REJECT;
+                        H5LTyylval.ival = atoi(yytext);
```

Comment:
Please use the HD* portability macros for all standard / system calls.  (atoi, strdup, etc)